contracts/ROSCA.sol

```diff
@@ -0,0 +1,222 @@
+  address winnerAddress;
+
+
+  modifier onlyForeman {
+  if(msg.sender != foreman) throw;
+  _;
+  }
+  modifier onlyBeforeStart {
+  if(currentRound != 0) throw;
+  _;
+  }
+
+  /**
+    * Creates a new ROSCA and initializes the necessary variables, ROSCA doesnt start until the specified startTime
+    * Creator of the contract becomes foreman and also added as the first member of the ROSCA
+    */
+  function ROSCA(
+    uint16 roundPeriodInDays_,
+    uint contributionSize_,
+    uint8 minParticipants_,
+    uint startTime_,
+    uint16 feeInThousandths_)
+  {
+    if (roundPeriodInDays < MIN_ROUND_PERIOD_IN_DAYS || roundPeriodInDays > MAX_ROUND_PERIOD_IN_DAYS) throw;
+    roundPeriodInDays = roundPeriodInDays_;
+    if (contributionSize_ < MIN_CONTRIBUTION_SIZE) throw;
+    contributionSize = contributionSize_;
+
+    numEpochs = EPOCH_TO_RUN;
+
+    if (minParticipants_ < MINIMUM_PARTICIPANTS) throw;// there should be at least 2 people to make a group
+    minParticipants = minParticipants_;
+
+    if (startTime_ < (now + MINIMUM_TIME_BEFORE_ROSCA_START)) throw;
+    startTime = startTime_;
+
+    if (feeInThousandths_ > MAX_FEE_IN_THOUSANDTHS) throw;
+    serviceFeeInThousandths = feeInThousandths_;
+
+
+    // set foreman
+    foreman = msg.sender;
+
+    // register foreman as a member of ROSCA and increment numRounds in a Epoch to 1
+    members[msg.sender] = User({paid: 0 , contributed: 0, alive: true, pendingWithdrawl: 0});
+    membersAddresses.push(msg.sender);
+  }
+
+  /** startRound() check if the round has ended, if so, get the winner of the currentRound's pot
+    * and add the amount to the winner's pending withdrawal
+    * if there were no bid during the round, address at selected at semi-random
+    *
+    * if currentRound = 0(ROSCA hasnt started), no winner is declared
+    */
+
+  function startRound()
+  {
+
+    if (now < startTime + (uint(currentRound + (currentEpoch * membersAddresses.length))  * (uint(roundPeriodInDays) * 1 days)) || membersAddresses.length < minParticipants)
+      throw;
+
+    if (currentRound != 0) {
+      if (winnerAddress == 0) //only true when there is no bidder in this round
+      {
+        // there is no bid in this round so find an unpaid address for this epoch
+        for (uint i = 0 ; i < membersAddresses.length; i++)
+        {
+          uint semi_random = now % membersAddresses.length;
+          if(members[membersAddresses[(currentRound-1 + i) % membersAddresses.length]].paid < currentEpoch)
+          winnerAddress = membersAddresses[currentRound-1 + i];
+          break;
+        }
+      }
+      if(members[winnerAddress].paid == currentEpoch) throw;
+      members[winnerAddress].pendingWithdrawl += lowestBid;
+      members[winnerAddress].paid++;
+      LogRoundFundsReleased(winnerAddress, lowestBid);
+    }
+    if (currentRound < membersAddresses.length)  // reset variables related to bidding
+    {
+      lowestBid = contributionSize * membersAddresses.length;
```

Comment:
you might want to add +1 to it, so that if just one person wants the money, then can get it without paying any fee.